### DISCUSSION TITLE
Best practice - Render vs ServeHTTP

### DISCUSSION THREAD

**Discussion by petmarsh** [a-h/templ#964] [General]
I was wondering, what would be the best practice in template rendering in http handlers (HandlerFunc) - use Render() on a templ.Component or do a templ.Handler(templ.Component).ServerHTTP()

With Render() there's probably more control over the resulting response, though also the need, at least for my setup with errcheck linter enabled, to check and handle the io.Writer error returned - then as the writer produces an error probably just log that as it's not possible to pass this info currently to the gui consumer (and the more ifs the more I get yelled at by the code checker about cognitive complexity and nesting)

templ.Handler and its ServeHTTP() seems to be intended to be used as handler directly, not within my own handlers, although quite a few projects I checked use this method within HandleFunc 

Any insights/thoughts/... on this are appreciated
Thanks

**Comment by a-h** [comment]
I tend to use `templ.Handler` within my own handlers.

The main benefit of `templ.Handler` is that it prevents partial rendering of failed templates (by default), and sets the HTML content type, which might not be set by Go if the text isn't recognised as HTML.

There's not much code, it's more of a convenience - https://github.com/a-h/templ/blob/main/handler.go

**Reply by petmarsh** [reply]
Thanks, how about for something like htmx's oob swaps with multiple components/templates in planned response - is then Render() a better option than ServeHTTP call for each template?
